src/server/src/main/java/io/cassandrareaper/storage/repairrun/CassandraRepairRunDao.java

```diff
@@ -461,19 +515,24 @@ private Collection<? extends RepairRun> getRepairRunsWithStateForCluster(
       RepairRun.RunState runState) {
 
     Collection<RepairRun> repairRuns = Sets.newHashSet();
-    List<ResultSetFuture> futures = Lists.newArrayList();
+    List<CompletionStage<AsyncResultSet>> futures = Lists.newArrayList();
 
     for (UUID repairRunId : clusterRepairRunsId) {
       futures.add(this.session.executeAsync(getRepairRunPrepStmt.bind(repairRunId)));
     }
 
-    for (ResultSetFuture future : futures) {
-      ResultSet repairRunResult = future.getUninterruptibly();
-      for (Row row : repairRunResult) {
-        repairRuns.add(buildRepairRunFromRow(row, row.getUUID("id")));
+    for (CompletionStage<AsyncResultSet> future : futures) {
+      AsyncResultSet results = future.toCompletableFuture().join();
+      while (true) {
+        for (Row row : results.currentPage()) {
+          repairRuns.add(buildRepairRunFromRow(row, row.getUuid("id")));
+        }
+        if (!results.hasMorePages()) {
```

Comment:
This is just nitpicking, but wouldn't this code be equal to

```java
do {
    for(Row row ...) { .. }
    results = results.fetchNextPage().toCompletableFuture().join();
} while(results.hasMorePages());
```

Reply:
Sure, I guess we should refactor those before cutting out a final v4.0.0 to use the driver consistently across all DAOs.